### ISSUE TITLE
strategy.builds can't create a class with a keyword argument named "target".

### ISSUE THREAD

**Issue by McBrainy** [HypothesisWorks/hypothesis-python#1104]
After a fresh `pip install hypothesis`, running this in the repl:

```python
from hypothesis import strategies as st
import collections
Something = collections.namedtuple('Something', ['target'])
st.builds(Something, target=st.integers())
```

results in the error:

```
TypeError: builds() got multiple values for keyword argument 'target'
```

The reason seems to be because `builds`'s first argument is called `target`. Now, in this example, I could have just written `st.builds(Something, st.integers())`, but some classes require keyword arguments. 

The only way I see of fixing this while maintaining backward compatibility would be to add a new method similar to builds that accepts a dict instead of `**kwargs`.

**Comment by DRMacIver** [comment]
Thanks for reporting this! It's something that came up on IRC a while back and I think we totally failed to turn it into an issue.

Currently the best workaround is to use builds with a lambda. Something like `builds(lambda not_called_target, **kwargs: MyClass(target=not_called_target, **kwargs), not_called_target=blah, ...`. It's a bit cumbersome but it works!

In terms of fixing this backwards compatibly, the following is actually a valid backwards compatible change:

```
def builds(hypothesis_internal_target=None, *args, **kwargs):
    if hypothesis_internal_target is None:
        hypothesis_internal_target = kwargs['target']
    ....
```

(modulo handling of error cases).

I'm currently a bit overloaded (I don't know if any of the others are in a better state right now) so unlikely to get to this any time soon given the existence of an easy workaround, but if you wanted to submit a PR around the above change we'd be very happy to accept and help you out on any messy details. If not, we'll definitely get to this eventually just no idea when!

**Comment by dchudz** [comment]
This should be resolved now (by https://github.com/HypothesisWorks/hypothesis-python/pull/1106). 

(Sorry, I guess maybe I should have said "fixes #1104" in the PR title so that happened automatically.)

**Comment by Zac-HD** [comment]
No worries!

(closing by keyword also works in the PR body text, for next time 😉)
